pkg/image/ostree_disk.go

```diff
@@ -26,21 +27,22 @@ type OSTreeDiskImage struct {
 	Users  []users.User
 	Groups []users.Group
 
-	CommitSource ostree.SourceSpec
+	CommitSource    *ostree.SourceSpec
+	ContainerSource *container.SourceSpec
 
 	SysrootReadOnly bool
 
 	Remote ostree.Remote
 	OSName string
+	Ref    string
 
 	KernelOptionsAppend []string
 	Keyboard            string
 	Locale              string
 
 	Filename string
 
-	Ignition         bool
-	IgnitionPlatform string
+	IgnitionPlatform *string
```

Comment:
Why make this a pointer instead of using empty string to indicate it isn't set?

Reply:
I don't really have a good answer.  I think I picked up a habit of making optionals `nil`-able but that's not a good reason.  You're right, an plain string with empty as the "off" value would work better and require less checks.

Reply:
mumble mumble `Option<String>`

Reply:
Done!